manual/manual/tutorials/objectexamples.etex

```diff
@@ -2,15 +2,20 @@
 \label{c:objectexamples}
 \pdfchapterfold{-15}{Tutorial: Objects in OCaml}
 %HEVEA\cutname{objectexamples.html}
-{\it (Chapter written by J�r�me Vouillon, Didier R�my and Jacques Garrigue)}
+{\it (Chapter written by J�r�me Vouillon, Didier R�my and Jacques Garrigue)}
 
 \bigskip
 
 \noindent This chapter gives an overview of the object-oriented features of
-OCaml. Note that the relation between object, class and type
-in OCaml is very different from that in mainstream
-object-oriented languages like Java or C++, so that you should not
-assume that similar keywords mean the same thing.
+OCaml.
+
+Note that the relationship between object, class and type in OCaml is different than
+in Java and C++, so you shouldn't assume that similar keywords mean the same thing.
+Object-oriented features are used much less frequently in OCaml than
+in mainstream object-oriented languages such as Java and C++.  OCaml has other
```

Comment:
The repetition of "such as Java and C++" is unnecessary here, IMO.

The word "other" is tautologous - "OCaml has alternatives"

Reply:
I made it:

```
Note that the relationship between object, class and type in OCaml is different than
in mainstream object-oriented languages such as Java and C++, so you shouldn't assume
that similar keywords mean the same thing.
Object-oriented features are used much less frequently in OCaml than
in those languages.  OCaml has other
alternatives that are often more appropriate, such as modules and functors.  Many
OCaml programs do not use objects at all.
```

From the dictionary: an alternative is "one of two or more available possibilities."  Using obect-oriented features is an alternative in OCaml.  "other alternatives" emphasizes excluding using object-oriented features.  But if you feel strongly I will drop "other".

Reply:
Feel strongly is perhaps strong (!), but it mentions one thing (the object system) and then refers to the alternatives. To refer to the "other alternatives", there should be a reference to at least one alternative between the reference to the object system and the "other alternatives". 🙂

I like your point of wanting emphasis, though - perhaps it could be made by saying "several/many alternatives"?

Reply:
Ah, I suspect we're not going to agree here so I will remove "other".  In the bigger picture it doesn't matter.

> To refer to the "other alternatives", there should be a reference to at least one alternative between the reference to the object system and the "other alternatives".

An alternative is "one of two or more available possibilities."  Using object-oriented features is implicitly an alternative in OCaml since they they "are used much less frequently in OCaml".

Reply:
Stepping in as a non-native speaker,  `alternative` has also a strong underlying connotation of `another choice`; and in `other alternatives` one may wonder if `other` is an emphatic redundancy or refers to yet another other possibility. I would prefer either `several alternatives` or `other possibilities` and avoid the potential ambiguity (and nuance) of the double negation in `other alternatives`.